packages/react-instantsearch-hooks-dom/src/ui/RefinementList.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+
+import { cx } from './lib/cx';
+
+import type { RefinementListItem } from 'instantsearch.js/es/connectors/refinement-list/connectRefinementList';
+
+export type RefinementListProps = React.HTMLAttributes<HTMLDivElement> & {
+  items: RefinementListItem[];
+  onRefine(item: RefinementListItem): void;
+  searchBox?: React.ReactNode;
+  noResults?: React.ReactNode;
+  showMoreButton?: React.ReactNode;
+};
+
+export function RefinementList({
+  items,
+  onRefine,
+  searchBox,
+  noResults,
+  showMoreButton,
+  ...props
+}: RefinementListProps) {
+  return (
+    <div {...props} className={cx('ais-RefinementList', props.className)}>
+      {searchBox && (
+        <div className="ais-RefinementList-searchBox">{searchBox}</div>
+      )}
+      {noResults ? (
+        <div className="ais-RefinementList-noResults">{noResults}</div>
+      ) : (
+        <ul className="ais-RefinementList-list">
+          {items.map((item) => (
+            <li
+              key={item.value}
+              className={cx(
+                'ais-RefinementList-item',
+                item.isRefined && 'ais-RefinementList-item--selected'
+              )}
+            >
+              <label className="ais-RefinementList-label">
+                <input
+                  checked={item.isRefined}
+                  className="ais-RefinementList-checkbox"
+                  type="checkbox"
+                  value={item.value}
+                  onChange={() => {
+                    onRefine(item);
+                  }}
+                />
+                <span
+                  className="ais-RefinementList-labelText"
+                  dangerouslySetInnerHTML={{ __html: item.highlighted! }}
```

Comment:
Could we use the `<Highlight>` component here?

Reply:
The `<Highlight>` component is still in review, so I can do this in another PR, once it's merged.

Reply:
It won't work as is, our `<Highlight>` component will only get its input from `.__highlightResult` and won't pass the value from `.highlighted`.

Reply:
We can always transform the hits structure before passing it to the component. We're doing this in some other libs.

Reply:
I agree, I find it better than handling this specific case in the highlighter.

Reply:
@dhayab It will work if we use the `<Highlight>` UI component (not the widget) which isn't coupled to this data shape.

Reply:
@Haroenv @dhayab Addressed in [fe10a4a](https://github.com/algolia/react-instantsearch/pull/3381/commits/fe10a4ac92bc68ba79cfca375ea520e239ab0833).